packages/repack/ios/ScriptManager.mm

```diff
@@ -218,8 +218,18 @@ - (void)downloadAndCache:(ScriptConfig *)config completionHandler:(void (^)(NSEr
   NSURLSessionDataTask *task = [[NSURLSession sharedSession]
       dataTaskWithRequest:request
         completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+          NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
+          NSInteger statusCode = [httpResponse statusCode];
           if (error != nil) {
             callback(error);
+          }else if(statusCode < 200 || statusCode >= 300){
+            NSDictionary *userInfo = @{
+                                          NSLocalizedFailureReasonErrorKey: [NSString stringWithFormat:@"Request should have returned with 200 HTTP status, but instead it received %ld", (long)statusCode]
+                                      };
+            NSError *httpError = [NSError errorWithDomain:NSURLErrorDomain
+                                                          code:statusCode
+                                                          userInfo:userInfo];
+            callback(httpError);
```

Comment:
these changes don't look like they belong in this PR - this looks like something we should definitely include though, but it would be best to also do this on Android. Can you open a new issue describing the problem and a PR with these changes? 🙏